src/analysis/normed_space/pi_Lp.lean

```diff
@@ -61,22 +65,50 @@ noncomputable theory
 
 variables {ι : Type*}
 
+/- These lemmas need to move.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -/
+
+-- this needs to go in `data.real.ennreal`
+instance : zero_le_one_class ℝ≥0∞ := ⟨zero_le 1⟩
```

Comment:
Really, this is about `canonically_ordered_...`, right?

Reply:
`canonically_ordered_add_monoid`?

Reply:
It's possible to put this instance on ℝ≥0∞, which implies `zero_le_one_class`. It turns out I don't need it for this PR though. I think it's not quite `derive`able in part because for some reason Lean doesn't infer the covariant classes for mul and add, so they are stated explicitly at the top of the file. I'll PR this separately.
```
noncomputable instance : linear_ordered_comm_monoid_with_zero ℝ≥0∞ :=
{ mul_le_mul_left := λ a b, mul_le_mul_left',
  zero_le_one := zero_le 1,
  .. ennreal.linear_ordered_add_comm_monoid_with_top,
  .. (show comm_semiring ℝ≥0∞, from infer_instance) }
```